app/views/searches/show.html.erb

```diff
@@ -5,7 +5,7 @@
     <p><%= @error_msg %></p>
   </div>
 <% end %>
-<%= link_to t("advanced_search"), advanced_search_path, class: "t-link--gray t-link--has-arrow" %>
+<%= link_to t("advanced_search"), params[:query].present? ? advanced_search_path(query: params[:query]) : advanced_search_path, class: "t-link--gray t-link--has-arrow" %>
```

Comment:
Would be `advanced_search_path((query: params[:query]).compact)` enough?

Reply:
You're right! `{query: params[:query]}` does the job. I've amended the commit and force-pushed it into my branch.